extensions/vscode/src/commands.ts

```diff
@@ -19,3 +20,30 @@ export const create = async (
 const getExecutableBinary = (context: ExtensionContext): string => {
   return path.join(context.extensionPath, "bin", "publisher");
 };
+
+export const openFileInEditor = async (
+  path: string,
+  options?: { selection?: EditConfigurationSelection },
+) => {
+  let start = new Position(0, 0);
+  let end = new Position(0, 0);
+
+  if (options && options.selection) {
+    start = new Position(
+      options.selection.start.line,
+      options.selection.start.character,
+    );
+    if (options.selection.end) {
+      end = new Position(
+        options.selection.end.line,
+        options.selection.end.character,
+      );
+    } else {
+      end = start;
+    }
+  }
+
+  await window.showTextDocument(Uri.file(path), {
+    selection: new Range(start, end),
```

Comment:
Always including the `selection` has a bit of an odd behavior when this is called without `options: { selection }` and the file is open. It moves the cursor up to `0:0` where previously the file just remained open.

Perhaps we can only call with `selection` if we were passed a selection?

Reply:
Sounds like a great idea. Done!